docs/middleware/op-submission.md

```diff
@@ -157,3 +157,27 @@ backend.use('apply', (request, next) => {
 
 {: .warn :}
 The `request.$fixup()` method may throw an error, which should be handled appropriately, usually by passing directly to the `next()` callback.
+
+## Comparing old snapshot version with new version
+
+Frequently, it becomes necessary to verify the changes made. This can be accomplished by leveraging two hooks, `apply` and `commit`, and creating a snapshot clone within the `apply` hook.
+
+```js
+// This is simple function that can deeply clone the object
+// Note that this function is not most efficient one
```

Comment:
I think no need to add commentary on efficiency; as @ericyhwang points out, it often depends on the exact shape of the thing you're cloning, and I don't think it's our place to determine this for people.

```suggestion
// Alternatively use your favourite deep-clone library
```